pkg/controllers/member/serviceexport/controller.go

```diff
@@ -256,6 +268,70 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) setAzureRelatedInformation(ctx context.Context, service *corev1.Service, export *fleetnetv1alpha1.InternalServiceExport) error {
+	export.Spec.Type = service.Spec.Type
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return nil
+	}
+	export.Spec.IsInternalLoadBalancer = service.Annotations[objectmeta.AzureLoadBalancerInternalAnnotation] == "true"
+	if export.Spec.IsInternalLoadBalancer {
+		// no need to populate the PublicIPResourceID and IsDNSLabelConfigured which are only applicable for external load balancer
+		return nil
+	}
+	id, err := r.fetchPublicIPResourceID(ctx, service)
+	if err != nil {
+		return err
+	}
+	if id == nil || len(*id) == 0 {
+		return fmt.Errorf("public ip resource id %v has not been configured in the load balancer yet", id)
```

Comment:
Does it mean cloud provider is still working on it? Do we retry? Do we check if service already has lbIP in status already?